src/SecurityChecks.js

```diff
@@ -0,0 +1,150 @@
+    const clp = field.classLevelPermissions;
+    const thisClassWarnings = clpWarnings[className] || [];
+    if (!clp) {
+      thisClassWarnings.push({
+        title: `No Class Level Permissions on ${className}`,
+        message:
+          'Class level permissions are a security feature from that allows one to restrict access on a broader way than the ACL based permissions. We recommend implementing CLPs on all database classes.',
+        link: 'https://docs.parseplatform.org/parse-server/guide/#class-level-permissions',
+      });
+      clpWarnings[className] = thisClassWarnings;
+      continue;
+    }
+    const keys = ['find', 'count', 'get', 'create', 'update', 'delete', 'addField'];
+    for (const key of keys) {
+      const option = clp[key];
+      if (className === '_User' && key === 'create') {
+        continue;
+      }
+      if (!option || option['*']) {
+        thisClassWarnings.push({
+          title: `Unrestricted access to ${key}.`,
+          message: `We recommend restricting ${key} on all classes`,
+          link: 'https://docs.parseplatform.org/parse-server/guide/#class-level-permissions',
+        });
+      } else if (Object.keys(option).length != 0 && key === 'addField') {
+        thisClassWarnings.push({
+          title: `Certain users can add fields.`,
+          message:
+            'Class level permissions are a security feature from that allows one to restrict access on a broader way than the ACL based permissions. We recommend implementing CLPs on all database classes.',
+          link: 'https://docs.parseplatform.org/parse-server/guide/#class-level-permissions',
+        });
+      }
+    }
+    clpWarnings[className] = thisClassWarnings;
+  }
+  return clpWarnings;
+}
+function ServerConfig(req) {
+  const options = req.config || req;
+  const warnings = [];
+  if (options.allowClientClassCreation) {
+    warnings.push({
+      title: 'Allow Client Class Creation is not recommended.',
+      message:
+        'Allow client class creation is not recommended for production servers it allows any user - authorized or not - to create a new class.',
+      link: 'https://docs.parseplatform.org/js/guide/#restricting-class-creation',
+    });
+  }
+  if (!options.masterKey.match('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{14,})')) {
+    warnings.push({
+      title: 'Weak masterKey.',
+      message:
+        'masterKey is a key that overrides all permissions. You should use a secure string for your masterKey',
+      link: 'https://docs.parseplatform.org/parse-server/guide/#keys',
+    });
+  }
+  let https = false;
+  try {
+    const serverURL = url.parse(options.serverURL);
+    https = serverURL.protocol === 'https:';
+  } catch (e) {
+    /* */
+  }
+  if (!https) {
+    warnings.push({
+      title: `Server served over HTTP`,
+      message: 'We strongly recommend using a HTTPS protocol.',
+    });
+  }
+  return warnings;
+}
+async function Files(req) {
```

Comment:
Please rename all functions to `checkXyz`, e.g. `checkDatabase`, to better describe what the function does.